plugins/woocommerce/src/Blocks/Utils/ProductCollectionUtils.php

```diff
@@ -92,4 +99,179 @@ private static function remove_empty_array_recursive( $array ) {
 		}
 		return $array;
 	}
+
+	/**
+	 * Parse editor's location context from WP Post.
+	 *
+	 * Possible contexts:
+	 * - post
+	 * - page
+	 * - single-product
+	 * - product-archive
+	 * - cart
+	 * - checkout
+	 * - product-catalog
+	 * - order-confirmation
+	 *
+	 * @param WP_Post $post The Post instance.
+	 *
+	 * @return string Returns the context.
+	 */
+	public static function parse_editor_location_context( $post ) {
+		$context = 'site';
+
+		if ( ! $post instanceof \WP_Post ) {
+			return $context;
+		}
```

Comment:
Similar to other comment. Do we need this if condition?

Reply:
Just because it's a global helper, and will produce a fatal error otherwise, it's more of a sanity check. Do you think it's better to strictly define the function and allow the error?

Reply:
Let's go with defensive programming as [mentioned](https://github.com/woocommerce/woocommerce/pull/46466#discussion_r1635961830) by Darren 🙌🏻